spec/controllers/users_controller_spec.rb

```diff
@@ -85,6 +85,50 @@
 
     context "its own profile" do
       describe "with valid params" do
+        context "communication opt-in" do
+          let(:now) { Time.new(2002, 10, 31) }
+          before do
```

Comment:
Only syntactic things, but I'd suggest to use the one line syntax here for these:
```ruby
before { Timecoop.freeze(now) }

after { Timecop.return }
```
Just because the file probably is already super huge...